frontend/src/utilities/useWatchHardwareProfiles.ts

```diff
@@ -0,0 +1,16 @@
+import { HardwareProfileKind } from '~/k8sTypes';
+import { CustomWatchK8sResult } from '~/types';
+import { groupVersionKind, HardwareProfileModel } from '~/api';
+import useK8sWatchResourceList from './useK8sWatchResourceList';
+
+export const useWatchHardwareProfiles = (
```

Comment:
If you are planning to use websocket to watch the hardware profiles, maybe also use it when fetching the hardware profile table. Otherwise, when user edits something on the cluster to something invalid, there is an error message on the nav item but nothing on the table because they are using different ways to fetch the hardware profiles.

<img width="1509" alt="Screenshot 2025-01-20 at 4 05 38 PM" src="https://github.com/user-attachments/assets/63923719-9d90-4c74-abfd-6fa4b55d8876" />